src/PhysicsEngine.cpp

```diff
@@ -179,6 +180,273 @@ void ForwardEuler::march_step(double tStamp, double tStep) {
+
+    Quaterniond orient = rocket_.get_q_ornt();
+
+    double inertia[9];  // moments of inertia
+    rocket_.get_I(inertia);
+
+    double mass = rocket_.get_mass();
+
+    /******************** Calculate Intermediate States **********************/
+    // Each state is used to calculate the next state
+
+    // Reformating of the initial state of the rocket
+    RungeKuttaState k1{pos_if, vel_if, accel_if, ang_vel_if, ang_accel_if};
+    // Calculated with Euler step using half tStep and initial state
+    RungeKuttaState k2 = calc_state(tStamp, 0.5 * tStep, k1);
+    // Calculated with Euler step using half tStep and k2 state
+    RungeKuttaState k3 = calc_state(tStamp + (0.5 * tStep), 0.5 * tStep, k2);
+    // Calculated with Euler step using full tStep and k3 state
+    RungeKuttaState k4 = calc_state(tStamp + (0.5 * tStep), tStep, k3);
+
+    /********************** Perform Runge-Kutta Method ************************/
+
+    // calculate weighted averages
+    Vector3d vel_avg = (k1.vel + (2 * k2.vel) + (2 * k3.vel) + k4.vel) / 6;
+    Vector3d accel_avg =
+        (k1.accel + (2 * k2.accel) + (2 * k3.accel) + k4.accel) / 6;
+    Vector3d ang_vel_avg =
+        (k1.ang_vel + (2 * k2.ang_vel) + (2 * k3.ang_vel) + k4.ang_vel) / 6;
+    Vector3d ang_accel_avg = (k1.ang_accel + (2 * k2.ang_accel) +
+                              (2 * k3.ang_accel) + k4.ang_accel) /
+                             6;
+
+    // calculate rocket data based on average values instaed of initial
+    pos_if += tStep * vel_avg;
+    vel_if += tStep * accel_avg;
+
+    Vector3d net_force_if = calc_net_force(tStamp, pos_if, vel_avg);
+    Vector3d net_torque_if = calc_net_torque(vel_avg, pos_if);
+
+    accel_if = net_force_if / mass;
+
+    ang_vel_if += tStep * ang_accel_avg;
+    ang_accel_if.x() = net_torque_if.x() / inertia[0];
+    ang_accel_if.y() = net_torque_if.y() / inertia[4];
+    ang_accel_if.z() = net_torque_if.z() / inertia[8];
+
+    orient = update_quaternion(orient, ang_vel_avg, tStep);
+
+    //---- Launch Rail ----
+    // very basic implementation
+    if (pos_if.norm() < 4.50) {
+        ang_vel_if.x() = 0;
+        ang_vel_if.y() = 0;
+        ang_vel_if.z() = 0;
+        ang_accel_if.x() = 0;
+        ang_accel_if.y() = 0;
+        ang_accel_if.z() = 0;
+    }
+
+    //---- Set Values ----
+    rocket_.set_r_vect(pos_if);
+    rocket_.set_r_dot(vel_if);
+    rocket_.set_r_ddot(accel_if);
+    rocket_.set_w_vect(ang_vel_if);
+    rocket_.set_w_dot(ang_accel_if);
+    rocket_.set_f_net(net_force_if);
+    rocket_.set_t_net(net_torque_if);
+    rocket_.set_q_ornt(orient);
+}
+
+/**
+ * @brief Takes in time and velocity to calculate net force
+ *
+ * @param tStamp Specific time stamp in the simulation
+ * @param vel_if Rocket's velocity with respect to the inertial frame
+ * @return Vector3  Vector containing the net force on the rocket in the x, y,
+ * and z directions
+ */
+Vector3d RungeKutta::calc_net_force(double tStamp, Vector3d pos_if,
```

Comment:
I think `calc_net_force` and `calc_net_torque` should be members of the `PhysicsEngine` base class and shared among the Euler and RK4 engines.

I don't think this blocks merging this PR, but please make a ticket for this if you agree.